### ISSUE TITLE
/usr/bin/env php --version is "php5.2" for me

### ISSUE THREAD

**Issue by pquerner** [netz98/n98-magerun#585]
I'd like to start the installation of magento using a different version of PHP, because the default "/usr/bin/env php --version" returns

```
PHP 5.2.17-pl0-gentoo (cli) (built: Nov 29 2012 18:37:11)
Copyright (c) 1997-2010 The PHP Group
Zend Engine v2.2.0, Copyright (c) 1998-2010 Zend Technologies
```

for me.

I'm calling the installation like to:
```
php5.5 ~/bin/n98-magerun.phar install --dbHost="XXXX" --dbUser="XXXXm" --dbPass="XXXX" --dbName="XXXXX" --useDefaultConfigParams=yes --installationFolder="." --baseUrl="http://XXXX.de/" --noDownload --forceUseDb="XXXX"
```

Its a managed server and I really can do anything by myself here. 

Thanks!

//Edit

```

#!/usr/bin/env n98-magerun.phar script
! echo ${php.version}
```

Returns ```5.5.14``` (call: ```php5.5 ~/bin/n98-magerun.phar script test.magerun```)

//Edit2
The exact error message is

```
PHP Parse error:  syntax error, unexpected T_FUNCTION in /xxx/src/app/code/core/Mage/Core/Helper/Abstract.php on line 240
```

Im just guessing that its PHP 5.2.

**Comment by cmuench** [comment]
@pquerner That because your default PHP is PHP 5.2.
Inside the install command we call the Magento installer with "/usr/bin/env" which resolves to your PHP 5.2 installation.
We could try to detect the currently executing PHP version and to use this one for installation.

**Comment by AydinHassan** [comment]
@cmuench We can just use `$_SERVER['_']`

**Comment by cmuench** [comment]
ok, then we should change the source.

**Comment by cmuench** [comment]
@AydinHassan We need a fallback for windows users. The "_" env variable is not set on this systems.

**Comment by AydinHassan** [comment]
@cmuench hmmm:
```
PHP_BINDIR (string)
Specifies where the binaries were installed into.
PHP_BINARY (string)
Specifies the PHP binary path during script execution. Available since PHP 5.4.
```

It's not looking so reliable

**Comment by cmuench** [comment]
We could try to use the constant. If it's not available (for PHP 5.3) we use "/usr/bin/env".

**Comment by AydinHassan** [comment]
Maybe we could do a check to see if  `PHP_BINDIR . '/php'` exists for less than `5.4`? I imagine most binaries will be named `php` then drop back to `/usr/bin/env`

**Comment by pquerner** [comment]
@cmuench As its a managed server I cannot change the "default cli" PHP Version without talking to the hoster guys. I can try tho. But it wouldnt help the general case.

@AydinHassan @cmuench 
Maybe this will help ?
http://stackoverflow.com/a/26251690/3020926

**Comment by tkn98** [comment]
I suggest to support  `PHP_BINARY` for the moment and if not available fall-back to env. If the default is PHP 5.2 then it is, somebody should change the environment (e.g. by setting up the path variable or aliasing) before executing magerun. There should be no need to reconfigure the whole system only to have the PHP command in the version you need.

Using `PHP_BINARY` however makes sense to me in any case and should fix the issue already for all PHP version not end of live.

**Comment by cmuench** [comment]
@tkn98 Checked this. Have a look on: http://3v4l.org/Z3klR

**Comment by tkn98** [comment]
@cmuench: looks good for the first part, `/usr/bin/env` is only necessary if that would be a shebang line, but this just executes the `php` command, doesn't it? It's looked up in the path anyway, so should not be necessary. Or not?

**Comment by ktomk** [comment]
Just seeing, thanks to a PR this has been fixed recently:  #752 - It's using` PHP_BINARY` and there is some fallback:
